simulation_ws/src/sagemaker_rl_agent/markov/environments/deepracer_env.py

```diff
@@ -74,17 +75,23 @@ def __init__(self):
 
         if node_type == SIMULATION_WORKER:
             # ROS initialization
-            self.ack_publisher = rospy.Publisher('/vesc/low_level/ackermann_cmd_mux/output',
-                                                 AckermannDriveStamped, queue_size=100)
-            self.racecar_service = rospy.ServiceProxy('/gazebo/set_model_state', SetModelState)
-            rospy.init_node('rl_coach', anonymous=True)
+            rclpy.init()
+            self.node = rclpy.create_node('rl_coach')
+            '''
+            Wheel controller used in ROS1 was not available in ROS2.
+            So used ackermann plugin here. It rquires velocity message of type Twist.
+            '''
+            # self.ack_publisher = self.node.create_publisher(AckermannDriveStamped, '/vesc/low_level/ackermann_cmd_mux/output', 10)
+            self.ack_publisher = self.node.create_publisher(Twist, '/racer_car/cmd_vel', 10)
+            self.racecar_service = self.node.create_client(SetEntityState, '/set_entity_state')
 
             # Subscribe to ROS topics and register callbacks
-            rospy.Subscriber('/progress', Progress, self.callback_progress)
-            rospy.Subscriber('/camera/zed/rgb/image_rect_color', sensor_image, self.callback_image)
-            self.world_name = rospy.get_param('WORLD_NAME')
+            self.node.create_subscription(Progress, '/progress', self.callback_progress, 10)
+            # self.node.create_subscription(sensor_image, '/camera/zed/rgb/image_rect_color', self.callback_image, 10)
+            self.node.create_subscription(sensor_image, '/zed_camera_left_sensor/image_raw', self.callback_image, 10)
+            self.world_name = os.environ.get("WORLD_NAME",'easy_track')
             self.set_waypoints()
-            self.aws_region = rospy.get_param('ROS_AWS_REGION')
+            self.aws_region = os.environ.get("ROS_AWS_REGION", "us-east-1")
```

Comment:
Why change the functionality of the node here? This can be set as a parameter form the env variable in the launch file

Reply:
@ryanewel,
I agree that here we have changed functionality of node. In ROS1 it was getting read from the node parameter. But "ROS_AWS_REGION" is not node configuration/specific parameter right ? So it doesn't seem logical to set/read it as a node config parameter.
Please give your suggestion on this.

Reply:
That makes sense. I'm fine with this direction.